Content.Server/_NF/GameRule/PointOfInterestSystem.cs

```diff
@@ -0,0 +1,272 @@
+using System.Numerics;
+using Content.Shared._NF.GameRule;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Content.Server.Maps;
+using Content.Server.Station.Systems;
+using Content.Shared._NF.CCVar; // Frontier
+using Robust.Shared.Configuration;
+using Content.Server.GameTicking.Presets;
+using Content.Server.GameTicking;
+using System.Linq;
+
+namespace Content.Server._NF.GameRule;
+
+/// <summary>
+/// This handles the dungeon and trading post spawning, as well as round end capitalism summary
+/// </summary>
+//[Access(typeof(NfAdventureRuleSystem))]
+public sealed class PointOfInterestSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+    [Dependency] private readonly MapLoaderSystem _map = default!;
+    [Dependency] private readonly MetaDataSystem _meta = default!;
+    [Dependency] private readonly StationSystem _station = default!;
+    [Dependency] private readonly StationRenameWarpsSystems _renameWarps = default!;
+    [Dependency] private readonly GameTicker _ticker = default!;
+
+    private List<Vector2> _stationCoords = new();
```

Comment:
If this is being stored in the system, because it's not associated with a game rule anymore, it'll need to be cleaned up on round end.